paasta_tools/contrib/service_shard_update.py

```diff
@@ -217,6 +268,40 @@ def main(args):
                         instance_config["cpus"] = args.cpus
                     if args.mem is not None:
                         instance_config["mem"] = args.mem
+                    if any(
+                        (
+                            args.autotune_min_cpus,
+                            args.autotune_max_cpus,
+                            args.autotune_min_mem,
+                            args.autotune_max_mem,
+                            args.autotune_min_disk,
+                            args.autotune_max_disk,
+                        )
+                    ):
+                        limit_config = {}
+                        limit_config["cpus"] = {
+                            "min": args.autotune_min_cpus,
+                            "max": args.autotune_max_cpus,
+                        }
+                        limit_config["mem"] = {
+                            "min": args.autotune_min_mem,
+                            "max": args.autotune_max_mem,
+                        }
+                        limit_config["disk"] = {
+                            "min": args.autotune_min_disk,
+                            "max": args.autotune_max_disk,
+                        }
+
+                        # remove any None values to keep the config clean
+                        for resource in list(limit_config):
+                            for key in list(limit_config[resource]):
```

Comment:
(these list()'s aren't necessary, but i'm assuming they're were helpful in some way while you were debugging and converting from a view to a list shouldn't really impact perf in a meaningful way for this script)

Reply:
unfortunately they are necessary under py3 :(

```
Traceback (most recent call last):
  File "paasta_tools/contrib/service_shard_update.py", line 350, in <module>
    main(args)
  File "paasta_tools/contrib/service_shard_update.py", line 297, in main
    for key in limit_config[resource]:
RuntimeError: dictionary changed size during iteration
```

Reply:
oh d'oh - i'm dumb: we're mutating the dict 

i blame friday-brain :p